pandas/core/indexers.py

```diff
@@ -324,8 +324,8 @@ def length_of_indexer(indexer, target=None) -> int:
             start, stop = stop + 1, start + 1
             step = -step
         return (stop - start + step - 1) // step
-    elif isinstance(indexer, (ABCSeries, ABCIndex, np.ndarray, list)):
-        if isinstance(indexer, list):
+    elif isinstance(indexer, (ABCSeries, ABCIndex, np.ndarray, list, range)):
+        if isinstance(indexer, (list, range)):
```

Comment:
the range case can be implemented separately + more efficiently, similar to the slice case

Reply:
You mean that a range object should have been converted to a slice before calling `length_of_indexer`?

At the moment it's simply the `_setitem_with_indexer_split_path` case that calls that method.

Reply:
Ah, but it's probably because of calling `self.iloc._setitem_with_indexer(..)` in frame.py instead of `self.iloc[..] = .. ` that the conversion doesn't happen anymore (`iloc.__setitem__` calls `indexer = self._get_setitem_indexer(key)` before passing the indexer to `_setitem_with_indexer` and the `_get_setitem_indexer` converts a range object in a list.

Reply:
i was just thinking `(indexer.start - indexer.stop) // indexer.step`

Reply:
does this not work?